pystiche_papers/sanakoyeu_et_al_2018/_augmentation.py

```diff
@@ -0,0 +1,492 @@
+            dct["interpolation"] = self.interpolation
+        if self.same_on_batch:
+            dct["same_on_batch"] = True
+        if self.align_corners:
+            dct["align_corners"] = True
+        return dct
+
+
+def affine_matrix_from_three_points(
+    points: torch.Tensor, transformed_points: torch.Tensor
+) -> torch.Tensor:
+    mat1 = transformed_points
+    mat2 = torch.inverse(torch.cat((points, torch.ones(points.size()[0], 1, 3)), dim=1))
+    return torch.bmm(mat1, mat2)
+
+
+class RandomAffine(AugmentationBase):
+    def __init__(
+        self,
+        shift: float,
+        probability: float = 50e-2,
+        interpolation: str = "bilinear",
+        same_on_batch: bool = False,
+        align_corners: bool = False,
+    ):
+        super().__init__(return_transform=False)
+        self.shift = shift
+        self.probability = probability
+        self.interpolation = interpolation
+        self.same_on_batch = same_on_batch
+        self.align_corners = align_corners
+
+    def generate_parameters(self, input_shape: torch.Size) -> Dict[str, torch.Tensor]:
+        batch_size = input_shape[0]
+        batch_params = cast(
+            Dict[str, torch.Tensor],
+            random_prob_generator(batch_size, self.probability, self.same_on_batch),
+        )
+
+        points = torch.tensor((((0, 0, 1), (0, 1, 0)),), dtype=torch.float).repeat(
+            batch_size, 1, 1
+        )
+        shift = _adapted_uniform(
+            points.size(), -self.shift, self.shift, same_on_batch=self.same_on_batch
+        )
+        shift[~batch_params["batch_prob"], ...] = 0.0
+
+        return {"matrix": affine_matrix_from_three_points(points, points + shift)}
+
+    def apply_transform(
+        self, input: torch.Tensor, params: Dict[str, torch.Tensor]
+    ) -> torch.Tensor:
+        return cast(
+            torch.Tensor,
+            F.warp_affine(
+                input,
+                params["matrix"],
+                extract_image_size(input),
+                flags=self.interpolation,
+                align_corners=self.align_corners,
+            ),
+        )
+
+    def _properties(self) -> Dict[str, Any]:
+        dct = super()._properties()
+        dct["shift"] = self.shift
+        dct["probability"] = f"{self.probability:.1%}"
+        if self.interpolation != "bilinear":
+            dct["interpolation"] = self.interpolation
+        if self.same_on_batch:
+            dct["same_on_batch"] = True
+        if self.align_corners:
+            dct["align_corners"] = True
+        return dct
+
+
+def _adapted_uniform_int(
+    shape: Union[Tuple, torch.Size],
+    low: Union[float, torch.Tensor],
+    high: Union[float, torch.Tensor],
+    same_on_batch: bool = False,
+) -> torch.Tensor:
+    return _adapted_uniform(shape, low, high + 1 - 1e-6, same_on_batch).int()
+
+
+class RandomCrop(AugmentationBase):
+    def __init__(
+        self,
+        size: Union[Tuple[int, int], int],
```

Comment:
The authors use a default `crop_size` of `(256,256)`. See [here](https://github.com/CompVis/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/img_augm.py#L26).

Reply:
That is true, but they are [not using the default](https://github.com/CompVis/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/model.py#L271) during training.